src/bindings/python/flux/accounting/job_archive_interface.py

```diff
@@ -218,3 +220,202 @@ def append_to_where(where_stmt, conditional):
+    ):
+        # fetch past usage factors
+        past_usage_factors = fetch_old_usage_factors(acct_conn, user, bank)
+
+        usage_user_historical = sum(past_usage_factors)
+    # if no new jobs were found but we are past the most recent half-life
+    # period, then the current usage needs to have a decay factor applied
+    # to it
+    elif len(job_totals_user) == 0 and (
+        float(t_end_half_life_period) < (time.time() - half_life_period)
+    ):
+        # fetch past usage factors
+        past_usage_factors = fetch_old_usage_factors(acct_conn, user, bank)
+
+        usage_user_historical = apply_decay_factor(0.5, acct_conn, user, bank)
+    # if last_t_inactive - t_end_half_life_period < half_life_period,
+    # append newly found jobs to the most recent usage factor
+    elif (last_t_inactive - float(t_end_half_life_period)) < half_life_period:
+        # append current usage to first usage factor bin
+        fetch_current_usage_factor = """
+            SELECT usage_factor_period_0
+            FROM job_usage_factor_table
+            WHERE username=?
+            AND bank=?
+            """
+        dataframe = pd.read_sql_query(
+            fetch_current_usage_factor, acct_conn, params=(user, bank,)
+        )
+        usage_factor_period_0 = dataframe.iloc[0]
+
+        usage_user_current += float(usage_factor_period_0)
+
+        # usage_user_past will just be the sum of the older factors since
+        # they will already had their decay factor applied to them in this
+        # current half-life period
+        usage_user_past = fetch_old_usage_factors(acct_conn, user, bank)
+
+        # calculate historical usage factor for user
+        usage_user_historical = usage_user_current + sum(usage_user_past[1:])
+    # else, create a new bin, move the older factors (and throw out the oldest
+    # one), and set a new end timestamp for the next half-life period
+    else:
+        # apply decay factor to past usage periods of a user's jobs
+        usage_user_past = apply_decay_factor(0.5, acct_conn, user, bank)
+
+        # calculate historical usage factor for user
+        usage_user_historical = usage_user_current + usage_user_past
+
+    # write last t_inactive to last_job_timestamp for user
+    update_timestamp_stmt = """
+        UPDATE job_usage_factor_table SET last_job_timestamp=?
+        WHERE username=?
+        AND bank=?
+        """
+    acct_conn.execute(update_timestamp_stmt, (last_t_inactive, user, bank,))
+    acct_conn.commit()
+
+    # write historical usage to first column in job_usage_factor_table
+    update_stmt = """
+        UPDATE job_usage_factor_table
+        SET usage_factor_period_0=?
+        WHERE username=?
+        AND bank=?
+        """
+    acct_conn.execute(update_stmt, (str(usage_user_historical), user, bank,))
+    acct_conn.commit()
+
+    return usage_user_historical
+
+
+def update_end_half_life_period(acct_conn, priority_decay_half_life):
```

Comment:
Where does this function get called from outside of the tests?  I'm not seeing it in any of the diffs.

Reply:
What I had in mind was that this function would get called by the cron job script (that isn't written yet) that runs every **PriorityCalcPeriod** that iterates through the user/bank combinations and calls `calc_usage_factor()` on each of those combinations. After it finishes iterating through those, it would call `update_end_half_life_period()` to see if we are in a new half-life period and update the appropriate tables.

I was thinking that in the next couple stages and PR's where we integrate `calc_usage_factor()` with `print-hierarchy`, Dong's fair tree library, write the `cron` job script, we would actually get to use and call `update_end_half_life_period()`. I hope that makes sense.

P.S Maybe I should document this in the **README.md** that talks about `calc_usage_factor()` and describe this workflow for that `cron` job script.

Reply:
>  I hope that makes sense.

Yeah. That does! Thanks for the explanation!

> P.S Maybe I should document this in the README.md that talks about calc_usage_factor() and describe this workflow for that cron job script.

That would be a great addition if you have the time.

Reply:
Not a problem - I went ahead and appended the above (hopefully in nicer wording 😆) to the **README.md** that outlines `calc_usage_factor()` in a `fixup` commit.